frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx

```diff
@@ -133,6 +169,8 @@ function MainNavbarView({
               selectedItem={cardItem ?? dashboardItem ?? collectionItem}
               onSelect={onItemSelect}
               reorderBookmarks={reorderBookmarks}
+              onToggle={_.debounce(setExpandBookmarks, debounceTimeout)}
```

Comment:
function will be re-initialized on every render. Can we throw this in a `useCallback`?

Reply:
Ooh good catch.

I think we probably always want to debounce the setter function, so I've now made the hook return a debounced version of the setter by default.